### PR TITLE
issue-163 : implemented OAM service methods

### LINKED ISSUE #163: Implement OAM Services for version 2.0.0
### **Changes to the existing OAM services:**
**TcpClient :**
- [x] Remove support for the getTcpClientRemoteIpv4Address , putTcpClientRemoteIpv4Address. Remove the functions from the controllers/TcpClient.js and service/TcpClientService.js
- [x] Include support for the APIs getTcpClientRemoteAddress , putTcpClientRemoteAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service
- [x] Include support for the APIs getTcpClientRemoteProtocol , putTcpClientRemoteProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service

**HttpClient :**
- [x] Include support for the API putHttpClientApplicationName . Copy the functions from the generated server stub , paste it to the controllers/HttpClient.js and service/HttpClientService.js. Implement the service

**TCPServer :**
- [x] Remove support for the getTcpServerLocalIpv4Address, putTcpServerLocalIpv4Address. Remove the functions from the controllers/TcpServer.js and service/TcpServerService.js
- [x] Include support for the APIs getTcpServerLocalAddress, putTcpServerLocalAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [x] Include support for the APIs getTcpServerLocalProtocol, putTcpServerLocalProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [x] Include support for the APIs getTcpServerDescription, putTcpServerDescription. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service

### **To include new functionalities of version 2.0.0 :**
**GenericResponseProfile :**
Support Version 2.0.0 changes for GenericResponseProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/GenericResponseProfile.js and service/GenericResponseProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support GenericResponseProfile in version 2.0.0
Implement the logic in the controllers and services for the following services , 
- [x] getGenericResponseProfileOperationName
- [x] getGenericResponseProfileFieldName
- [x] getGenericResponseProfileDescription
- [x] getGenericResponseProfileDatatype
- [x] getGenericResponseProfileValue
- [x] putGenericResponseProfileValue

**ActionProfile :**
Support Version 2.0.0 changes for ActionProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/ActionProfileProfile.js and service/ActionProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support ActionProfileProfile in version 2.0.0
Implement the logic in the controllers and services for the following services ,
- [x] getActionProfileOperationName
- [x] getActionProfileLabel
- [x] getActionProfileInputValueListt
- [x] getActionProfileDisplayInNewBrowserWindow
- [x] getActionProfileConsequentOperationReference
- [x] putActionProfileConsequentOperationReference

Core :
Include support for the API getProfileInstance. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service.

### To include new Individual functionalities in version 2.0.0 :

IntegerProfile :
Support Version 2.0.0 changes for IntegerProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/IntegerProfile.js and service/IntegerProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code

Following tasks needs to be completed to support IntegerProfile in version 2.0.0
Implement the logic in the controllers and services for the following services , 
- [ ] getIntegerProfileIntegerName
- [ ] getIntegerProfileUnit
- [ ] getIntegerProfileMinimum
- [ ] getIntegerProfileMaximum
- [ ] getIntegerProfileIntegerValue
- [ ] putIntegerProfileIntegerValue - Configuration range should be within the /integer-profile-capability/minimum and /integer-profile-capability/maximum

ElasticsearchClient :
- [x] https://github.com/openBackhaul/ExecutionAndTraceLog/issues/108

### PR DESCRIPTION
**Pull request by DanaSunal**
Fixes #163

### CODE REVIEW

**Comment by DanaSunal** [comment]
Reviewer is @vanithavalluripalli9

File: server/service/HttpClientService.js [hunk 1]
```diff
@@ -50,6 +49,22 @@ exports.getHttpClientReleaseNumber = function (url) {
...
+ * Configures name of application to be addressed
+ *
+ * body Httpclientinterfaceconfiguration_applicationname_body
+ * uuid String
+ * no response value expected for this operation
+ **/
+exports.putHttpClientApplicationName = function(body, url) {
+  return new Promise(async function (resolve, reject) {
+    try {
+      await fileOperation.writeToDatabaseAsync(url, body, false);
```
Comment (vanithavalluripalli9) [inline server/service/HttpClientService.js:62]: Please add forwarding automation to ALT

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]
Please automate forwardings to ALT for the PUT operations of HttpClient , TcpClient , OperationClient , TcpServer , HttpServer , OperationServer as we are updating them.

File: server/service/HttpClientService.js (hunk 1 above)
Comment (DanaSunal) [reply server/service/HttpClientService.js]: Done.

**Pull request merged**